feature: Separate SMTP Credentials

JediKev opened the pull request:
This feature adds the ability to separate the IMAP/POP and SMTP credentials for a System Email. This adds a checkbox next to the Authentication Required setting where, if Enabled, will show two additional inputs for the SMTP Username and Password. If Authentication Required is set to No the SMTP Username and Password inputs will disappear as they are not needed. If the checkbox is Disabled the system will use the IMAP/POP credentials for SMTP.

**Disabled:**
![unchecked](https://user-images.githubusercontent.com/11823401/67330698-607b7280-f4e2-11e9-9189-29a7cc429a34.png)

**Enabled:**
![checked](https://user-images.githubusercontent.com/11823401/67330692-5bb6be80-f4e2-11e9-9b05-95805cbcf9d9.png)

JediKev committed:
feature: Fetch From Mail Folder

This feature adds a textbox in the Email configurations that allows you to
configure a specific Folder to fetch from. If no folder is provided the
system will fetch from the INBOX as usual. This also adds a check so that if
a Folder is given but doesn't exist and can't be created then the system
will error out with a detailed message.

JediKev committed:
feature: Separate SMTP Credentials

This feature adds the ability to separate the IMAP/POP and SMTP credentials
for a System Email. This adds a checkbox next to the Authentication Required
setting where, if Enabled, will show two additional inputs for the SMTP
Username and Password. If Authentication Required is set to No the SMTP
Username and Password inputs will disappear as they are not needed. If the
checkbox is Disabled the system will use the IMAP/POP credentials for SMTP.

JediKev committed:
issue: Authentication Autocomplete

This addresses an annoying issue where Chrome autofills the Username and
Password for Email Authentication even though we have the
`autocomplete="off"` attribute set. This changes the attribute from
`autocomplete="off"` to `autocomplete="new-password"` so that Chrome will
not attempt to autofill the credentials. More information on this can be
found [here](https://bugs.chromium.org/p/chromium/issues/detail?id=468153).

JediKev committed:
patch: Add Mark Answered Permission

This adds the Mark Answered permission by default to Roles with the Reply
Permission Enabled. Piggy backs on 5135.

protich commented on include/class.mailfetch.php:
```diff
@@ -126,14 +126,18 @@ function getMaxFetch() {
     }
 
+    function getMailFolder() {
+        return $this->mailbox_encode($this->ht['folder']) ?: 'INBOX';
```
I know mailbox_encode returns a null on null folder...  but that's not guaranteed forever should the logic get modified...  something like below will make me comfortable. 

```php
return $this->mailbox_encode($this->ht['folder'] ?: 'INBOX');
```

JediKev replied:
Done

protich commented on include/staff/email.inc.php:
```diff
@@ -324,9 +334,28 @@
                      <?php echo !$info['smtp_auth']?'checked':''; ?> /> <?php echo __('No'); ?></label>
+                 &nbsp;
+                 <label><input type="checkbox" name="smtp_auth_creds" value="1"
+                     <?php echo $info['smtp_auth_creds']?'checked':''; ?> />Use Separate Authentication</label>
```
Put a space before and make text translatable.

JediKev replied:
Done

JediKev committed:
css: Checkbox Vertical Alignment

This addresses a small issue where the system shows inline checkboxes
centered at the bottom of the text which makes it look out of place. This
updates the CSS for `input[type=checkbox]` to make the `vertical-align`
property equal to `middle` which renders the checkbox correctly.

JediKev committed:
issue: Add Default Embed Whitelist Domains

This adds the default Domains for the Embed Whitelist feature.

protich committed:
Merge branch 'develop-next' into feature/separate-smtp-creds
